command/agent/cache/handler.go

```diff
@@ -0,0 +1,127 @@
+package cache
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"time"
+
+	"github.com/hashicorp/errwrap"
+	hclog "github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/vault/api"
+	"github.com/hashicorp/vault/helper/consts"
+	vaulthttp "github.com/hashicorp/vault/http"
+	"github.com/hashicorp/vault/logical"
+)
+
+type Config struct {
+	Client           *api.Client
+	Proxier          Proxier
+	UseAutoAuthToken bool
+	Listeners        []net.Listener
+	Logger           hclog.Logger
+}
+
+func Run(ctx context.Context, config *Config) error {
```

Comment:
Can `config` be `nil`? Maybe add a check like you did at https://github.com/hashicorp/vault/pull/6220/files#diff-703e1234dc09e66de48735d23f23732eR90

Reply:
Fixed.